icechunk/src/storage/caching.rs

```diff
@@ -44,11 +51,27 @@ impl MemCachingStorage {
             chunk_cache: Cache::new(num_chunks as usize),
         }
     }
+
+    pub fn new_with_defaults(backend: Arc<dyn Storage + Send + Sync>) -> Self {
+        Self::new(backend, 2, 2, 0, 2, 0)
+    }
 }
 
 impl private::Sealed for MemCachingStorage {}
 
+impl<'de> Deserialize<'de> for MemCachingStorage {
+    fn deserialize<D>(deserializer: D) -> Result<MemCachingStorage, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        let backend: Arc<dyn Storage + Sync + Send> =
+            Deserialize::deserialize(deserializer)?;
+        Ok(Self::new_with_defaults(backend))
```

Comment:
This is quite dangerous ... we should preserve the cache sizes during serialization and recover them

Reply:
I think this is what we are doing currently right? I dont think we serialize the Cacheing Storage at all?

Reply:
True, good point ... let me create a ticket